Sources/Interaction/Manipulators/MouseCameraUnicamRotateManipulator/index.js

```diff
@@ -0,0 +1,368 @@
+    // the border of the screen on each side
+    const borderRatio = 0.1;
+    // If someone has already clicked to make a dot and they're not clicking
+    // on it now, OR if the user is clicking on the perimeter of the screen,
+    // then we want to go into rotation mode.
+    if (
+      Math.abs(normalizedPosition.x) > 1 - borderRatio ||
+      Math.abs(normalizedPosition.y) > 1 - borderRatio ||
+      model.isDot
+    ) {
+      model.state = States.IS_ROTATE;
+    } else {
+      model.state = States.IS_NONE;
+    }
+  };
+
+  //----------------------------------------------------------------------------
+  publicAPI.onMouseMove = (interactor, renderer, position) => {
+    if (!model.buttonPressed) {
+      return;
+    }
+
+    model.state = States.IS_ROTATE;
+    rotate(interactor, position);
+
+    model.previousPosition = position;
+  };
+
+  //--------------------------------------------------------------------------
+  publicAPI.onButtonUp = (interactor) => {
+    const renderer = interactor.findPokedRenderer();
+    model.buttonPressed = false;
+
+    if (model.state === States.IS_ROTATE && model.isDot) {
+      renderer.removeActor(model.focusSphere);
+      model.isDot = false;
+    } else if (model.state === States.IS_NONE) {
+      if (model.isDot) {
+        renderer.removeActor(model.focusSphere);
+        model.isDot = false;
+      } else {
+        model.focusSphere.setPosition(...model.downPoint);
+
+        const camera = renderer.getActiveCamera();
+        const cameraPosition = camera.getPosition();
+        const cameraToPointVec = [];
+
+        vtkMath.subtract(model.downPoint, cameraPosition, cameraToPointVec);
+        if (camera.getParallelProjection()) {
+          vtkMath.multiplyScalar(cameraToPointVec, camera.getParallelScale());
+        }
+
+        const atV = camera.getDirectionOfProjection();
+        vtkMath.normalize(atV, interactor);
+
+        // Scales the focus dot so it always appears the same size
+        const scale =
+          0.02 *
+          vtkMath.dot(atV, cameraToPointVec) *
+          model.focusSphereRadiusFactor;
+
+        model.focusSphere.setScale(scale, scale, scale);
+
+        renderer.addActor(model.focusSphere);
+
+        model.isDot = true;
+      }
+    }
+
+    renderer.resetCameraClippingRange();
+    updateAndRender(interactor);
+  };
+
+  publicAPI.getFocusSphereColor = () => {
+    model.focusSphere.getProperty().getColor();
+  };
+  publicAPI.setFocusSphereColor = (r, g, b) => {
+    model.focusSphere.getProperty().setColor(r, g, b);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  focusSphereRadiusFactor: 1,
+  useHardwareSelector: true,
+  useWorldUpVec: true,
+  // set WorldUpVector to be z-axis by default
+  worldUpVec: [0, 0, 1],
```

Comment:
I do not see the 2 modes:
 - down&drag to rotate
 - click then drag to rotate

Reply:
These two interactions are already available in this manipulator, but I did not define two different modes because they are not overlapping. I can split it in two modes if you think there is a point in doing so.

Reply:
nope, sounds good to me. thanks.